streams/src/test/java/org/apache/kafka/streams/kstream/KStreamBuilderTest.java

```diff
@@ -233,4 +236,54 @@ public void shouldMapStateStoresToCorrectSourceTopics() throws Exception {
         assertEquals(Collections.singletonList("table-topic"), builder.stateStoreNameToSourceTopics().get("table-store"));
         assertEquals(Collections.singletonList(APP_ID + "-KSTREAM-MAP-0000000003-repartition"), builder.stateStoreNameToSourceTopics().get("count"));
     }
-}
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void kStreamTimestampExtractorShouldBeNull() throws Exception {
+        builder.stream("topic");
+        final ProcessorTopology processorTopology = builder.build(null);
+        assertNull(processorTopology.source("topic").getTimestampExtractor());
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void kStreamShouldAddTimestampExtractorWithKeyValSerdePerSource() throws Exception {
+        builder.stream(new MockTimestampExtractor(), null, null, "topic");
+        final ProcessorTopology processorTopology = builder.build(null);
+        for (final SourceNode sourceNode: processorTopology.sources()) {
+            assertTrue(sourceNode.getTimestampExtractor() instanceof MockTimestampExtractor);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void kStreamhouldAddTimestampExtractorWithOffsetResetPerSource() throws Exception {
```

Comment:
typo: kStreamhould... -> kStreamShould
In fact i'd probably rename these methods to begin with should, i.e., 
`shouldAddTimestampExtractorToStreamWithOffsetResetPerSource` etc